syntax/problem.php

```diff
@@ -147,18 +164,69 @@ private function renderFigures(Doku_Renderer &$renderer, \PluginFKSTaskRepo\Task
         }
     }
 
+    /**
+     * Outputs no image files
+     * @param Doku_Renderer $renderer
+     * @param \PluginFKSTaskRepo\Task $data
+     */
+    private function renderFileAttachments(Doku_Renderer &$renderer, \PluginFKSTaskRepo\Task $data) {
+        if (is_array($data->getFigures())) {
+            $renderer->doc .= '<div class="task-fileattachments">';
+            foreach ($data->getFigures() as $figure) {
+                if($this->isImage(ml($figure['path']))) continue; // Checks if it is an image
+                $renderer->doc .= '<div class="task-fileattachments-file">';
+                $renderer->internalmedia($figure['path'], $figure['caption'] ?: null,null,null,null,null,'linkonly');
+                $renderer->doc .= '</div>';
+            }
+            $renderer->doc .= '</div>';
+        }
+    }
+
     private function renderTags(Doku_Renderer &$renderer, \PluginFKSTaskRepo\Task $data) {
         $tags = $this->helper->loadTags($data->getYear(), $data->getSeries(), $data->getLabel());
         foreach ($tags as $tag) {
             $renderer->doc .= $this->helper->getTagLink($tag, null, $data->getLang());
         }
     }
 
+    private function renderOrigin(Doku_Renderer &$renderer, \PluginFKSTaskRepo\Task $data) {
+        $renderer->doc .= '<div class="origin">' . $renderer->render_text($data->getOrigin()) . '</div>';
+    }
+
     private function renderTask(Doku_Renderer &$renderer, \PluginFKSTaskRepo\Task $data) {
-        // TODO trim is very ugly
         $renderer->doc .= '<div>' . $renderer->render_text(trim($data->getTask())) . '</div>';
     }
 
+    /**
+     * Renders links to PDF with solution to specific task. Czech PDF is rendered always though it is on english site.
+     * @param Doku_Renderer $renderer
+     * @param \PluginFKSTaskRepo\Task $data
+     * @return bool If solution exists
+     */
+    private function renderSolutions(Doku_Renderer &$renderer, \PluginFKSTaskRepo\Task $data) {
+        global $conf;
+
+        $path = vsprintf($this->getConf('solution_path_' . $conf['lang']), [$data->getYear(), $data->getSeries(), $data->getLabel()]); // Add path
+        $path = file_exists(mediaFN($path)) ? $path : null;
+
+        // Include original cs PDF to en (if exists obviously)
+        $original = vsprintf($this->getConf('solution_path_cs'), [$data->getYear(), $data->getSeries(), $data->getLabel()]); // Add path
+        $original = file_exists(mediaFN($original)) && $conf['lang'] != 'cs' ? $original : null;
```

Comment:
prečo nie !== ?

Reply:
Nechápu

Reply:
prečo používaš len != namiesto !==?

Reply:
Nechápu, proč bych neměl. Vždyť v tomto případě to je jedno, ne?

Reply:
napr. pre code quality? Bude to mať plne očakávateľne správanie, a by-default sa používa !== (===) len v prípade že potrebujem porovnávať hodnoty nezávisle na typoch používam != (==)